src/controllers/components/model.js

```diff
@@ -1,12 +1,30 @@
-import { searchModel } from '../../lib/api-client';
+import { searchModel, deleteModel, getAllModels } from '../../lib/api-client';
 import { Dispatcher } from 'consus-core/flux';
 
 export default class ModelController {
     static getModel(address) {
         return searchModel(address).then( model => {
-            Dispatcher.handleAction("MODEL_FOUND", model);
+            Dispatcher.handleAction('MODEL_FOUND', model);
         }).catch( () => {
-            Dispatcher.handleAction("NO_MODEL_FOUND");
+            Dispatcher.handleAction('NO_MODEL_FOUND');
+        });
+    }
+
+    static deleteModel(address){
+        return deleteModel(address).then(data => {
+            Dispatcher.handleAction('MODEL_DELETED', data.deletedModel);
+        }).catch(error => {
+            Dispatcher.handleAction('ERROR', {
+                error: error.message
+            });
+        }).then(() => {
+            return getAllModels().then(models => {
```

Comment:
We should break this promise chain up.

Rather than

```javascript
asyncStuff().then(() => {
    return asyncWhatever();
}).then(() => {
    return asyncThing().then(() => {
        return moreAsyncStuff();
    }).catch(e => {
        dealWith(e);
    });
});
```

We can/should build on the original chain, like this:

```javascript
asyncStuff().then(() => {
    return asyncWhatever();
}).then(() => {
    return asyncThing();
}).then(() => {
    return moreAsyncStuff();
}).catch(e => {
    dealWith(e);
});
```

Promises are used to avoid the deep indentations/nesting.